src/libCli/GrammarConstruction.cpp

```diff
@@ -57,41 +56,42 @@ class GrammarInjectorMethodArgs : public GrammarInjector
 
             //std::cout << f_parseTree->getDebugString() << std::endl;
             //std::cout << "Injecting grammar for " << serverAddress << ":" << serverPort << " " << serviceName << " " << methodName << std::endl;
-            std::shared_ptr<grpc::Channel> channel = ConnectionManager::getInstance().getChannel(serverAddress);
+            std::shared_ptr<grpc::Channel> channel = ConnectionManager::getInstance().getChannel(serverAddress, *(f_parseTree));
```

Comment:
brackets not required
```suggestion
            std::shared_ptr<grpc::Channel> channel = ConnectionManager::getInstance().getChannel(serverAddress, *f_parseTree);
```